src/lib.rs

```diff
@@ -228,15 +128,47 @@ impl CrashReporter {
         );
         data.insert("version".to_string(), json!(report.version));
 
-        if let Some(threshold_value) = &self.threshold {
-            data.insert("threshold".to_string(), json!(threshold_value));
+        if data
+            .get("errorGroups")
+            .unwrap()
+            .as_array()
+            .unwrap()
+            .is_empty()
+        {
+            let mut template = Handlebars::new();
+            template
+                .register_template_string(
+                    "no_crashes_found_template",
+                    self.no_crashes_found_template(),
+                )
+                .expect("Failed to register the no crashes found template.");
+            return template
+                .render("no_crashes_found_template", &json!(data))
+                .unwrap();
+        }
+
+        // add threshold
+        if let Some(threshold) = self.threshold {
+            self.add_threshold_values_to_individual_crashes(data, threshold);
+            data.insert("threshold".to_string(), json!(threshold));
+        }
+
+        if self.use_arithmetic_mean {
```

Comment:
Won't this also execute even if the crash reports are empty?

Reply:
The return statement in [line 145](https://github.com/xing/recrep/pull/10/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R145) is essentially an early return version of [line 172](https://github.com/xing/recrep/pull/10/files#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R172). So all following `if something`s are skipped